lib/network/httputils/json.go

```diff
@@ -49,7 +48,13 @@ func WriteJSON(w http.ResponseWriter, code int, v interface{}) error {
 
 	w.WriteHeader(code)
 
-	bs, err := json.Marshal(v)
+	var bs []byte
+	var err error
+	if marshaler, ok := v.(json.Marshaler); ok {
+		bs, err = marshaler.MarshalJSON()
```

Comment:
`json.Marshal` already takes care of that if you implement the interface

Reply:
By using json.Marshal method, it impossible to set escapeHTML = false
As you can see here, Line 456 opts.escapeHTML is set true by default.
https://github.com/golang/go/blob/22dbc96d8a004bb78a0efd329d0f48b6f0a8b9e5/src/encoding/json/encode.go#L443-L461

Reply:
It is not set to `true` by default, it's just passing the value that was set in the option.
So if you set it to `false` in the caller, it should just work.

Reply:
the `compact` function overwrite the option.

I think it is a bug of the json library. @Geod24